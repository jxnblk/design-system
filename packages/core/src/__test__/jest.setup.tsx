import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
