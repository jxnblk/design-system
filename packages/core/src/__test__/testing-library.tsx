import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
