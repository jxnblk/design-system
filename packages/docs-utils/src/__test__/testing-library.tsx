import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
