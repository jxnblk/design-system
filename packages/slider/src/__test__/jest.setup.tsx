import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
