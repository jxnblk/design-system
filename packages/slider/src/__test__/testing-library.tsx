import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
