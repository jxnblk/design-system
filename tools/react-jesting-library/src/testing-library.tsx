import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
