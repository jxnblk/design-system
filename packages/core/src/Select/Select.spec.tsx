import React from 'react'
import { render, screen } from '../__test__/testing-library'

import { Select, theme, createTheme, getPaletteColor } from '..'

describe('Select', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<Select />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('it renders disabled', () => {
    const paletteTheme = createTheme(theme)
    render(
      <Select defaultValue='Premium Economy' disabled>
        <option>Premium Economy</option>
      </Select>
    )

    const select = screen.getByText('Premium Economy').closest('select')
    expect(select).toBeDisabled()
    expect(select).toHaveStyleRule(
      'background-color',
      getPaletteColor('background.light')({ theme: paletteTheme }),
      { modifier: ':disabled' }
    )
    expect(select).toHaveStyleRule('color', getPaletteColor('text.light')({ theme: paletteTheme }), {
      modifier: ':disabled',
    })
    expect(select).toHaveStyleRule('cursor', 'not-allowed', { modifier: ':disabled' })
    expect(select).toHaveStyleRule('opacity', '1', { modifier: ':disabled' })
  })

  test('it renders read-only', () => {
    const paletteTheme = createTheme(theme)
    render(
      <Select defaultValue='Premium Economy' readOnly>
        <option>Premium Economy</option>
      </Select>
    )

    const select = screen.getByText('Premium Economy').closest('select')

    expect(screen.getByText('Premium Economy')).toBeDisabled()
    expect(select).toHaveAttribute('readonly')
    expect(select).toHaveStyleRule(
      'background-color',
      getPaletteColor('background.light')({ theme: paletteTheme }),
      { modifier: '[readOnly]' }
    )
    expect(select).toHaveStyleRule('color', getPaletteColor('text.light')({ theme: paletteTheme }), {
      modifier: '[readOnly]',
    })
    expect(select).toHaveStyleRule('cursor', 'not-allowed', { modifier: '[readOnly]' })
    expect(select).toHaveStyleRule('opacity', '1', { modifier: '[readOnly]' })
  })
})
