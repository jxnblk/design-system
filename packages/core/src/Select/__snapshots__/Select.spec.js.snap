// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-left: -32px;
  width: 24px;
  color: #4f6f8f;
}

.c3 {
  outline: none;
  pointer-events: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #c0cad5;
  margin: 0;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 32px;
  font-size: 16px;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c1::-ms-expand {
  display: none;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 14px;
  }
}

<div
  className="c0"
  width={1}
>
  <select
    className="c1 "
    fontSize={
      Array [
        2,
        null,
        1,
      ]
    }
  />
  <svg
    aria-hidden={true}
    className="c2 c3"
    color="text.light"
    fill="currentcolor"
    focusable={false}
    height={24}
    role="img"
    tabIndex={-1}
    viewBox="0 0 24 24"
    width={24}
  >
    <path
      d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
    />
  </svg>
</div>
`;
